presto-pinot-toolkit/src/test/java/com/facebook/presto/pinot/query/TestPinotQueryGeneratorSql.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.pinot.query;
+
+import com.facebook.presto.testing.assertions.Assert;
+import org.testng.annotations.Test;
+
+public class TestPinotQueryGeneratorSql
+        extends TestPinotQueryGenerator
+{
+    @Override
+    public boolean useSqlSyntax()
+    {
+        return true;
+    }
+
+    @Test
+    public void assertUsingSqlSyntax()
+    {
+        Assert.assertEquals(defaultSessionHolder.getConnectorSession()
+                .getProperty("use_pinot_sql_for_broker_queries", Boolean.class).booleanValue(), true);
+    }
+
+    @Override
+    public String getExpectedAggOutput(String expectedAggOutput, String groupByColumns)
```

Comment:
I didn't follow this: I thought in the SQL end point, its okay to have a Pinot SQL query like: "select count(1), A, B, Sum(C) from Table group by B, A" ... ie the group by columns's don't have to come before the agg columns. (And thus we don't need `expectedColumnIndices`). Is that case tested ?

I am curious what is the pinot SQL expected for these presto sql queries:
- "select A, B, Count(1), Sum(C) from Table group by B, A"
-"select Count(1), A, Sum(C), B from Table group by B, A"

Reply:
This one works. Pinot handles the selection expression ordering and return the results following that.

Reply:
This method is to override the expected selection clause in the query.
E.g. assume PQL is `select Count(1), Sum(C) from Table group by B, A`, then in SQL the query will be `select B, A, Count(1), Sum(C) from Table group by B, A` in order to get same results.

Reply:
Okay, so this implementation of `getExpectedAggOutput` does not allow mixing of agg and group by columns. It does handle the group by columns to be explicitly present (which is not the case for PQL). But how would the Presto SQL query with mixed group by and aggregation get translated in the SQL endpoint: "select count(1), A, Sum(C), B from Table group by B, A" ? (note that the aggregations and group by columns are intermingled here and you can't clump the two separately)

Reply:
The logic is that AggregationNode always have full aggregation/groupby info, and a ProjectNode will handle the assignment. So we can push down another field, say `outputColumns`, to handle the logic of what to be pushed down in selection.

Reply:
Also `SELECT A from Table group by A` passes only `TableScanNode` to `PinotPlanOptimizer`, I will look into this maybe in next PR.